### PR TITLE
add skew-normal distribution and tests

### PR DESCRIPTION
**Pull request by aloctavodia**
I think this is a potentially useful distribution that was missing from PyMC3.

### CODE REVIEW

**Review by springcoil — APPROVED** [review]
Excellent - I think this is useful. If this passes the tests - we can merge.

**Comment by aloctavodia** [comment]
Weird! All tests pass locally, but not here.

**Comment by ColCarroll** [comment]
Same test that failed on #1381 -- not sure why it would fail now (the random seed should be set just before it runs).  I wonder if the threads get the same seed?

**Comment by ColCarroll** [comment]
Looks like they _should_: http://stackoverflow.com/questions/9209078/using-python-multiprocessing-with-different-random-seed-for-each-process

**Comment by aloctavodia** [comment]
So, this is just a random failure, right? As a dirty workaround, can we just re-run travis?

**Comment by ColCarroll** [comment]
If the seed is set correctly, you should get the exact same failure, so
it'd at least be interesting!

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -1237,3 +1238,66 @@ def logp(self, value):
...
+    Skew-normal distribution can be parameterized either in terms of precision
+    or standard deviation. The link between the two parametrizations is
+    given by
+    .. math::
+       \tau = \dfrac{1}{\sigma^2}
+    Parameters
+    ----------
+    mu : float
+        Location parameter.
+    tau : float
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: Can you replace the order of tau and sd to make it match the calling signature?

Comment (aloctavodia) [reply pymc3/distributions/continuous.py]: Sure, thanks!

**Comment by twiecki** [comment]
Thanks @aloctavodia!

**Pull request merged**